features/bonding/examples/mco-bonding.md

```diff
@@ -0,0 +1,72 @@
+# Bonding using machine-config-operator (MCO)
+
+Project page [machine-config-operator](https://github.com/openshift/machine-config-operator)
+
+This document focusses on using `Machine Config operator (MCO)` for interface bonding and vlan configurations on a node. `MachineConfig` CRD is used to write configuration files onto the node. It expects the file contents to be supplied as encoded data per current ignition specifications, the encoding is done via `base64` in this document.
+
+### Sample bond interface file
+
+```
+BONDING_OPTS="miimon=140 mode=balance-rr"
+TYPE=Bond
+BONDING_MASTER=yes
+BOOTPROTO=dhcp
+NAME=bond1
+DEVICE=bond1
+ONBOOT=yes
+AUTOCONNECT_SLAVES=yes
+```
+
+### Encoding using base64
+
+```
+cat ifcfg-bond1 | base64
+```
+
+### Machine config pool
+
+The machineconfigpool object shows a group of nodes to which a machine config can be applied.
+
+```
+$ oc get machineconfigpools
+NAME     CONFIG                                             UPDATED   UPDATING   DEGRADED   MACHINECOUNT   READYMACHINECOUNT   UPDATEDMACHINECOUNT   DEGRADEDMACHINECOUNT   AGE
+master   rendered-master-ddc67ca664d7c26305129dd659483226   True      False      False      3              3                   3                     0                      18h
+worker   rendered-worker-6c00884cabc50f4d42433e4e3acd603c   True      False      False      1              1                   1                     0                      18h
+```
+
+### Machine Config
+
+This configuration creates a MachineConfig to write the `bond1` interface configuration file at the path location `/etc/sysconfig/network-scripts/ifcfg-bond1` on the worker nodes. Multiple files can be specified here.
+ 
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: bonding-test-config
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,<encoded data>
+        filesystem: root
+        mode: 0644
+        path: /etc/sysconfig/network-scripts/ifcfg-bond1
+```
+
+### Node configuration examples
+
+```
+10: bond1: <BROADCAST,MULTICAST,MASTER,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
+    link/ether XX:XX:XX:XX:XX:XX brd ff:ff:ff:ff:ff:ff
+    inet 192.X.X.100/24 brd 192.X.X.255 scope global noprefixroute bond1
+       valid_lft forever preferred_lft forever
+11: bond1.20@bond1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
```

Comment:
with the example this interface will not be create, can you please change the example to create two interfaces?

Reply:
Good catch. done!